crates/pessimistic-proof-core/src/proof.rs

```diff
@@ -142,87 +154,69 @@ pub const EMPTY_LER: Digest = Digest(hex!(
-        batch_header.target.nullifier_root.as_slice(),
-        batch_header.target.ler_leaf_count.to_le_bytes().as_slice(),
-    ]);
-
-    let mut network_state = initial_network_state;
-    let computed_target = network_state.apply_batch_header(batch_header)?;
-
-    if computed_target.exit_root != batch_header.target.exit_root {
-        return Err(ProofError::InvalidNewLocalExitRoot {
-            declared: batch_header.target.exit_root,
-            computed: computed_target.exit_root,
-        });
-    }
-
-    if computed_target.balance_root != batch_header.target.balance_root {
-        return Err(ProofError::InvalidNewBalanceRoot {
-            declared: batch_header.target.balance_root,
-            computed: computed_target.balance_root,
-        });
-    }
-
-    if computed_target.nullifier_root != batch_header.target.nullifier_root {
-        return Err(ProofError::InvalidNewNullifierRoot {
-            declared: batch_header.target.nullifier_root,
-            computed: computed_target.nullifier_root,
-        });
-    }
+    let target_pp_root_version = initial_network_state.verify_consensus(batch_header)?;
+
+    let new_pessimistic_root = {
+        let mut network_state = initial_network_state;
+        let computed_target = network_state.apply_batch_header(batch_header)?;
+
+        if computed_target.exit_root != batch_header.target.exit_root {
+            return Err(ProofError::InvalidNewLocalExitRoot {
+                declared: batch_header.target.exit_root,
+                computed: computed_target.exit_root,
+            });
+        }
+
+        if computed_target.balance_root != batch_header.target.balance_root {
+            return Err(ProofError::InvalidNewBalanceRoot {
+                declared: batch_header.target.balance_root,
+                computed: computed_target.balance_root,
+            });
+        }
+
+        if computed_target.nullifier_root != batch_header.target.nullifier_root {
+            return Err(ProofError::InvalidNewNullifierRoot {
+                declared: batch_header.target.nullifier_root,
+                computed: computed_target.nullifier_root,
+            });
+        }
+
+        let Some(height) = batch_header.height.checked_add(1) else {
+            return Err(ProofError::HeightOverflow);
+        };
 
-    if computed_target.ler_leaf_count != batch_header.target.ler_leaf_count {
-        return Err(ProofError::InvalidNewLocalExitRootLeafCount {
-            declared: batch_header.target.ler_leaf_count,
-            computed: computed_target.ler_leaf_count,
-        });
-    }
+        PessimisticRoot {
+            balance_root: network_state.balance_tree.root,
+            nullifier_root: network_state.nullifier_tree.root,
+            ler_leaf_count: network_state.exit_tree.leaf_count,
+            height,
+            origin_network: batch_header.origin_network,
+        }
+        .compute_pp_root(target_pp_root_version)
```

Comment:
It should always be `v3` here right?

Reply:
after migration yes (i.e., right after that the aggsender adds the height in its signature), but before that it'll stay v2